### ISSUE TITLE
Problem with installing on Azure: Ubuntu16.04

### ISSUE THREAD

**Issue by J-Moravec** [Azure/blobxfer#56]
I am trying to install this on Azure's virtual machine, but I am getting an error regarding associated library click.

```
Traceback (most recent call last):
  File "/usr/local/bin/blobxfer", line 11, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 676, in main
    _verify_python3_env()
  File "/usr/local/lib/python3.5/dist-packages/click/_unicodefun.py", line 118, in _verify_python3_env
    'for mitigation steps.' + extra)
RuntimeError: Click will abort further execution because Python 3 was configured to use ASCII as encoding for the environment.  Consult http://click.pocoo.org/python3/for mitigation steps.

This system supports the C.UTF-8 locale which is recommended.
You might be able to resolve your issue by exporting the
following environment variables:

    export LC_ALL=C.UTF-8
    export LANG=C.UTF-8

```

Trying to export locale does not solve this issue.

Can't reproduce locally.

Also, I tried to use pip instead of pip3, but it seems like it didn't had any effect?

```
#!/bin/bash
set -e

export LANG=en_US.utf8
export LC_ALL=en_US.utf8

sudo apt-get update -y
sudo apt-get install -y build-essential libssl-dev libffi-dev libpython3-dev python3-dev python3-pip
sudo pip3 install --upgrade pip
sudo pip3 install --upgrade azure-batch azure-storage msrestazure # blobxfer
# coz the click library used by blobxfer is bad
sudo pip install --upgrade pip
sudo pip install --upgrade blobxfer
```

**Comment by J-Moravec** [comment]
This error is because the default locale is set to POSIX. Exporting locale helps locally (heh), but not globally in all scripts. Weird thing is that /etc/default/locale is en_US.UTF-8.

**Comment by alfpark** [comment]
I cannot reproduce your installation issue on an Azure allocated Ubuntu 16.04 VM. It works out of the box with:

```
sudo apt-get update
sudo apt-get install -y build-essential libssl-dev libffi-dev python3-dev python3-pip
sudo pip3 install blobxfer
```

Users, typically by default, will have a default locale set. If for some reason you don't, you need to set them.

**Comment by J-Moravec** [comment]
This was done through program that was given to us to make setting up jobs easier (if there is a way how to upload jobs without creating all those JSON templates, I would be grateful!). So as per our mail communication, yes, this is through Azure Batch, so the default locale was POSIX. Because of that, inside of sudo pip3, the locale reseted to POSIX (for some reason). But since whoami returned root, I could just install without sudo (or try --user, but that brings another set of problems)...

So the program we are using set up to run the command line: job.sh > job.log 2>&1; blobxfer ... so it run outside of scripts. So even exported environment in all scripts couldn't work. I tried to add command with exporting environment, but then... blobxfer would fail on call (as incorrectly called).

My interpretation was that if I don't export locale, it will work (correct call), but it will fail while using the click library somewhere inside. But when I add locale, it suddenly treat it as wrong call.

What I totally missed was that it could be incorrect call in both cases and the click library fails just quicker. So what I totally missed is that the last time I used these scripts was much later, than I thought, and that the API totally changed in the meantime.

So I think I am good now, sorry for bothering you!
